### ISSUE TITLE
1.1.x compilation / packaging issues

### ISSUE THREAD

**Issue by 64616E69656C** [tatsuhiro-t/nghttp2#298]
Hi,

with v1.1.0 / v1.1.1 I'm running into two smaller (but annoying) issues.
*Manual* compilation has to be done with autoreconf -i && automake && autoconf first, because I'm on automake 1.15 and nghttp2 expects 1.14. 
```
CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash /home/daniel/code/nghttp2/nghttp2-1.1.1.afl/missing aclocal-1.14 -I m4
/home/daniel/code/nghttp2/nghttp2-1.1.1.afl/missing: line 81: aclocal-1.14: command not found
WARNING: 'aclocal-1.14' is missing on your system.
```

Another thing, I had to --disable-python-bindings completly, because it does not read PYTHONPATH from env, configure or anything else but "sudo PYTHONENV=/other/path make install", not to mention debuild were I have not found a workaround yet (when installing into $(CURDIR)/debian/tmp/usr/lib/python2.7/site-packages).

With 1.1.1, another issue popped during packaging:
```
clang++-3.6 -DHAVE_CONFIG_H -I. -I..  -Wall -I../lib/includes -I../lib/includes -I../src/includes -I../third-party -DHAVE_CONFIG_H -pthread -I/usr/include -D_FORTIFY_SOURCE=2  -O2 -g2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -fstack-protector -fPIE -std=c++11 -c -o asio_sv-asio-sv.o `test -f 'asio-sv.cc' || echo './'`asio-sv.cc
/bin/bash ../libtool  --tag=CXX   --mode=link clang++-3.6  -O2 -g2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -fstack-protector -fPIE -std=c++11  -Wl,-z,relro -fPIE -pie -o asio-sv asio_sv-asio-sv.o ../lib/libnghttp2.la ../src/libnghttp2_asio.la -ljemalloc ../third-party/libhttp-parser.la -L/usr/lib/x86_64-linux-gnu -lboost_system -lboost_thread -lboost_system -lssl -lcrypto -pthread
libtool: link: clang++-3.6 -O2 -g2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security -fstack-protector -fPIE -std=c++11 -Wl,-z -Wl,relro -fPIE -pie -o .libs/asio-sv asio_sv-asio-sv.o -pthread  ../lib/.libs/libnghttp2.so ../src/.libs/libnghttp2_asio.so -ljemalloc ../third-party/.libs/libhttp-parser.a -L/usr/lib/x86_64-linux-gnu -lboost_thread -lboost_system -lssl -lcrypto -pthread
../src/.libs/libnghttp2_asio.so: undefined reference to `nghttp2_timegm'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:567: recipe for target 'asio-sv' failed
make[3]: *** [asio-sv] Error 1
```

Edit: also happens when manually compiling:
```
  CXXLD    asio-sv
afl-cc 1.83b by <lcamtuf@google.com>
../src/.libs/libnghttp2_asio.so: undefined reference to `nghttp2_timegm'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:576: recipe for target 'asio-sv' failed
```

Oh, and also this happend on v1.1.0, but I accidently enabled -Werror once :-)
```
configure:17038: clang-3.6 -o conftest -Werror -O2 -g2 -Wformat -Wformat-security -Werror=format-security -fstack-protector -fPIE -D_FORTIFY_SOURCE=2 -I/usr/include/libxml2 -Wl,-z,relro -fPIE -pie conftest.c -lxml2 -L/usr/lib/x86_64-linux-gnu -lz   >&5
conftest.c:39:3: error: ignoring return value of function declared with warn_unused_result attribute [-Werror,-Wunused-result]
system("touch conf.xmltest");
^~~~~~ ~~~~~~~~~~~~~~~~~~~~
```

**Comment by tatsuhiro-t** [comment]
Thank you for reporting this.
I believe nghttp2_timegm issue was fixed with commit c470ac7b0021d3cae80ef1c5b6460a108f2e5bdb

Python installation issue, it seems setuptools is very picky wrt installation path (https://pythonhosted.org/setuptools/easy_install.html#custom-installation-locations).  There is pretty much chance that I may not understand its functionality yet.
The following workaround would make python module install where we specified in --prefix:
```diff
diff --git a/python/Makefile.am b/python/Makefile.am
index 7a35239..4da5228 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -33,7 +33,8 @@ all-local: nghttp2.c
 	$(PYTHON) setup.py build
 
 install-exec-local:
-	$(PYTHON) setup.py install --prefix=$(DESTDIR)$(prefix)
+	test -d $(pyexecdir) || mkdir -p $(pyexecdir)
+	PYTHONUSERBASE=$(DESTDIR)/$(prefix) $(PYTHON) setup.py install --user
 
 uninstall-local:
 	rm -f $(DESTDIR)$(libdir)/python*/site-packages/nghttp2.so
```

setuptools also creates or edits easy-install.pth under site-packages, which could be annoying for packages.  If this works for you, I'll merge it into master. 

For libtool issue, ubuntu 15.04 still automake 1.14, so as debian.  As far as I know, you only need `autoreconf -i` once per edits to libtool stuff.  Not so big deal?

**Comment by 64616E69656C** [comment]
Thank you very much, I can confirm, timegm patch for Makefile.am works just fine.

Also, libtool is of course not an annoying issue (that's python :), dh-autoreconf solved this.

I'll play around with the python workaround after upgrading my systems to 1.1.1, since I personally don't need the python-bindings anyway. Therefore I'm closing this.
